apps/prairielearn/src/lib/uri-util.shared.ts

```diff
@@ -13,3 +15,19 @@ export function encodePathNoNormalize(originalPath: string): string {
     return '';
   }
 }
+
+/**
+ * Encodes a set of query parameters into a URI component string to be used with search query strings.
+ * @param entries Record of query parameter names and values
+ * @returns Encoded query parameter string
+ */
+export function encodeQueryParams(entries: Record<string, string | null | undefined>): string {
```

Comment:
Nit: we (and a lot of the web ecosystem) use "query params" to mean generically the key/values at the end of the URL. This is not using it in that sense, this is very specific to this `search-string` library.

Should we call this something like `encodeSearchString` instead?